interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -1759,37 +1761,68 @@ public void testOrcaPerRpc() throws Exception {
    */
   public void testOrcaOob() throws Exception {
     AtomicReference<TestOrcaReport> reportHolder = new AtomicReference<>();
-    TestOrcaReport answer = TestOrcaReport.newBuilder()
-        .setCpuUtilization(0.8210)
-        .setMemoryUtilization(0.5847)
-        .putUtilization("util", 0.30499)
+    Random r = new Random();
+    final TestOrcaReport answer = TestOrcaReport.newBuilder()
+        .setCpuUtilization(r.nextDouble())
+        .setMemoryUtilization(r.nextDouble())
+        .putUtilization("util", r.nextDouble())
         .build();
-    blockingStub.unaryCall(SimpleRequest.newBuilder().setOrcaOobReport(answer).build());
-    int i;
-    int retryLimit = 5;
-    for (i = 0; i < retryLimit; i++) {
+    final TestOrcaReport answer2 = TestOrcaReport.newBuilder()
+        .setCpuUtilization(r.nextDouble())
+        .setMemoryUtilization(r.nextDouble())
+        .putUtilization("util", r.nextDouble())
+        .build();
+
+    final int retryLimit = 5;
+    BlockingQueue<StreamingOutputCallResponse> queue = new LinkedBlockingQueue<>();
+    SettableFuture<Void> closeFuture = SettableFuture.create();
+
+    StreamObserver<StreamingOutputCallRequest> streamObserver =
+        asyncStub.fullDuplexCall(new StreamObserver<StreamingOutputCallResponse>() {
+
+          @Override
+          public void onNext(StreamingOutputCallResponse value) {
+            queue.add(value);
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            closeFuture.setException(new IllegalStateException("unexpected stream error", t));
```

Comment:
`queue.add(t)` here and `queue.add(null)` for onCompleted(). Otherwise in the case of error the `queue.take()` will block forever and there won't be any useful error message.